src/lib/dlt_user.c

```diff
@@ -445,11 +506,17 @@ DltReturnValue dlt_init_file(const char *name)
 DltReturnValue dlt_init_message_queue(void)
 {
     dlt_lock_mutex(&mq_mutex);
+#ifdef DLT_TRACE_LOAD_CTRL_ENABLE
+    dlt_lock_mutex(&check_trace_load_mutex); // Ignore return value
```

Comment:
When I try to run a simple test with dlt-example-user and WITH_DLT_TRACE_LOAD_CTRL=ON, the example application is pending. Note: Flag is on but there is no LIMIT or BUDGET set.
``dlt-example-user -l 4 Hello``
Since while sending a dlt message, you are checking trace load and inside that function, you are taking the lock of mutex.
So I think you should remove lock of mutex check_trace_load_mutex here.